erpnext/stock/doctype/stock_entry/stock_entry.py

```diff
@@ -690,6 +691,25 @@ def update_stock_ledger(self):
 
 		self.make_sl_entries(sl_entries, self.amended_from and 'Yes' or 'No')
 
+	def validate_mandatory_batch_number(self):
+		# to validate the stock items batch number is present or not
+		items_without_batch_no = []
+		MANDATE_MSG = 'Batch number is mandatory for Item at : <br><b>'
+		item_list = self.get("items")
+		
+		for index in range(len(item_list)):
+			item_code_list = self.items[index].item_code
```

Comment:
if this is a single item, then just use item_code. Avoid using list for a string variable